packages/react-chat/src/components/Modals/WalletModal.tsx

```diff
@@ -9,14 +16,87 @@ import { WalletConnectLogo } from "../Icons/WalletConnectLogo";
 import { CoinbaseModalName } from "./CoinbaseModal";
 import { Modal } from "./Modal";
 import { Heading, MiddleSection, Section, Text } from "./ModalStyle";
+import { UserCreationModalName } from "./UserCreationModal";
 import { WalletConnectModalName } from "./WalletConnectModal";
 
 export const WalletModalName = "WalletModal";
 
 export function WalletModal() {
   const { setModal } = useModal(WalletModalName);
+  const setIdentity = useSetIdentity();
+  const setWalletIdentity = useSetWalletIdentity();
+  const userCreationModal = useModal(UserCreationModalName);
   const { setModal: setWalleConnectModal } = useModal(WalletConnectModalName);
   const { setModal: setCoinbaseModal } = useModal(CoinbaseModalName);
+  const { messenger } = useMessengerContext();
+
+  const handleMetamaskClick = useCallback(async () => {
+    const ethereum = (window as any)?.ethereum as any | undefined;
+    if (ethereum && messenger) {
+      try {
+        if (ethereum?.isMetaMask) {
+          const [account] = await ethereum.request({
+            method: "eth_requestAccounts",
+          });
+
+          const msgParams = JSON.stringify({
+            domain: {
+              chainId: 1,
+              name: window.location.origin,
+              version: "1",
+            },
+            message: {
+              action: "Status Chat Key",
+              onlySignOn: "https://auth.status.im/",
+              message:
+                "I'm aware that i am signing message that creates a private chat key for status communicator. And I have double checked everything is fine.",
```

Comment:
Instead:

`This signature will be used to decrypt chat communications. Check that the `onlySignOn` property of this message matches the chat website address`